tests/test_sink_and_static.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+import falcon
+from falcon import testing
+
+from _util import create_app  # NOQA
+
+
+def sink(req, res, **kw):
+    res.body = 'sink'
+
+
+async def sink_async(req, res, **kw):
+    res.body = 'sink'
+
+
+@pytest.fixture
+def client(asgi, tmp_path):
+    file = tmp_path / 'file.txt'
+    file.write_text('foo bar')
+
+    def make(sink_before_static_route):
+        app = create_app(asgi=asgi, sink_before_static_route=sink_before_static_route)
+        app.add_sink(sink_async if asgi else sink, '/sink')
```

Comment:
Do we need separate sinks here?
IIRC there was a way to invoke sync responders asynchronously if some envvars were present, however, maybe that applies only to resources and not sinks? :thinking:

Reply:
Looking at this quickly I think I think the asgi does require them to be async. In this case is seems we are missing documentation and test regarding sink/static routing for asgi.

Reply:
It is documented as [`wrap_sync_to_async_unsafe`](https://falcon.readthedocs.io/en/latest/api/util.html#falcon.wrap_sync_to_async_unsafe).

The [``FALCON_ASGI_WRAP_NON_COROUTINES``](https://github.com/falconry/falcon/blob/master/falcon/util/sync.py#L161) envvar is probably too hacky for public consumption, it is mostly there to DRY our tests.

Reply:
(Regardless, this is not required to pass review.)

Reply:
sinks and static routes (iirc) are added without further checks. I'll open an issue improve the docs and implement the checks

Reply:
see #1789